internal/controller/clusterinstance/template_engine.go

```diff
@@ -42,6 +43,92 @@ type TemplateEngine struct {
 	Log logr.Logger
 }
 
+type Action string
+
+const (
+	Render   Action = "render"
```

Comment:
I typically like to name these with the type if it's not clear from the package name.

So they would be `ActionRender`, `ActionSuppress`, `ActionPrune`.